docker-app/qfieldcloud/core/tests/test_secret_packages.py

```diff
@@ -0,0 +1,245 @@
+            self.u1,
+            self.p1,
+            "DCIM/1.jpg",
+            io.FileIO(testdata_path("DCIM/1.jpg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/2.jpg",
+            io.FileIO(testdata_path("DCIM/2.jpg"), "rb"),
+        )
+
+        # create secrets for project and assigned to users
+        self._create_secret(name="SECRET", project=self.p1, value="p1")
+        self._create_secret(
+            name="SECRET", project=self.p1, assigned_to=self.u1, value="u1"
+        )
+        self._create_secret(
+            name="SECRET", project=self.p1, assigned_to=self.u2, value="u2"
+        )
+
+        self.p1.refresh_from_db()
+
+        # create a package for the two users
+        repackage(self.p1, self.u1)
+        repackage(self.p1, self.u2)
+
+        wait_for_project_ok_status(self.p1)
+        self.p1.refresh_from_db()
+
+        latest_package_jobs_qs = self.p1.latest_package_jobs()
+
+        self.assertEquals(latest_package_jobs_qs.count(), 2)
+
+        latest_packages_users = latest_package_jobs_qs.values_list(
+            "created_by", flat=True
+        )
+
+        self.assertIn(self.u1.id, latest_packages_users)
+        self.assertIn(self.u2.id, latest_packages_users)
+
+    def test_create_org_level_packages(self):
+        # upload data & QGIS project files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "bumblebees.gpkg",
+            io.FileIO(testdata_path("bumblebees.gpkg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "simple_bumblebees.qgs",
+            io.FileIO(testdata_path("simple_bumblebees.qgs"), "rb"),
+        )
+
+        # upload some attachments files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/1.jpg",
+            io.FileIO(testdata_path("DCIM/1.jpg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/2.jpg",
+            io.FileIO(testdata_path("DCIM/2.jpg"), "rb"),
+        )
+
+        # create secrets for organization and assigned to user u1
+        self._create_secret(name="SECRET", organization=self.o1, value="o1")
+        self._create_secret(
+            name="SECRET", organization=self.o1, assigned_to=self.u1, value="u1"
+        )
+
+        self.p1.refresh_from_db()
+
+        # create a package for user u1
+        repackage(self.p1, self.u1)
+
+        wait_for_project_ok_status(self.p1)
+        self.p1.refresh_from_db()
+
+        latest_package_jobs_qs = self.p1.latest_package_jobs()
+
+        self.assertEquals(latest_package_jobs_qs.count(), 1)
+
+        # create a package for user u2
+        repackage(self.p1, self.u2)
+
+        latest_package_jobs_qs = self.p1.latest_package_jobs()
+
+        self.assertEquals(latest_package_jobs_qs.count(), 2)
```

Comment:
See my comment here for improved testing. Now you are just checking the count, but not the actual contents of the returning query. https://github.com/opengisch/QFieldCloud/pull/1235/files#r2167570659